dockstore-webservice/src/main/java/io/dockstore/webservice/core/database/EntryLite.java

```diff
@@ -16,142 +19,118 @@
  * @author ldcabansay
  * @since 1.9.0
  */
-public abstract class EntryLite {
+public abstract class EntryLite<S extends Entry<?, ?>> {
     private final Date lastUpdated;
-
-    EntryLite(Date entryUpdated, Date versionUpdated) {
-        //choose the greater update time between overall entry and most recently updated version
-        this.lastUpdated = versionUpdated == null ? entryUpdated : (entryUpdated.getTime() > versionUpdated.getTime() ? entryUpdated : versionUpdated);
+    private final EntryTypeMetadata entryTypeMetadata;
+    private final String entryPath;
+    private final String trsId;
+
+    EntryLite(Entry<?, ?> entry, Date entryUpdated, Date versionUpdated) {
```

Comment:
This might be slightly more correct if the first argument was of type `S`:
```
EntryLite(S entry, Date entryUpdated, Date versionUpdated) {
```
Hard to tell if that'll break anything else until making the change and building.  Not sure it's worth a lot of effort, tho, probably ok as is.